src/gamma_inc.jl

```diff
@@ -941,17 +941,23 @@ function __gamma_inc_inv(a::Float64, minpq::Float64, pcase::Bool)
     logr = (logp + loggamma1pa) / a
     if logr < log(0.2*(1 + a)) #small value of p
         x0 = gamma_inc_inv_psmall(a, logr)
-    elseif !pcase && minpq < min(0.02, exp(-1.5*a)/gamma(a)) && a < 10 #small q
-        x0 = gamma_inc_inv_qsmall(a, minpq)
-    elseif abs(minpq - 0.5) < 1.0e-05
-        x0 = a - 1.0/3.0 + (8.0/405.0 + 184.0/25515.0/a)/a
-    elseif abs(a - 1.0) < 1.0e-4
-        x0 = pcase ? -log1p(-minpq) : -log(minpq)
-    elseif a < 1.0 # small value of a
-        x0 = exp(logr)
-    else    #large a
-        haseta = true
-        x0, fp = gamma_inc_inv_alarge(a, minpq, pcase)
+    else
+        # This deviates from the original version. The tmp variable
+        # here ensures that the argument of sqrt in gamma_inc_inv_qsmall
+        # is positive
+        tmp = inv(gammax(a)*sqrt(twoπ/a))
+        if !pcase && minpq < min(0.02, tmp) && a < 10 #small q
+            x0 = gamma_inc_inv_qsmall(a, minpq)
+        elseif abs(minpq - 0.5) < 1.0e-05
+            x0 = a - 1.0/3.0 + (8.0/405.0 + 184.0/25515.0/a)/a
+        elseif abs(a - 1.0) < 1.0e-4
+            x0 = pcase ? -log1p(-minpq) : -log(minpq)
+        elseif a < 1.0 # small value of a
+            x0 = exp(logr)
+        else    #large a
+            haseta = true
+            x0, fp = gamma_inc_inv_alarge(a, minpq, pcase)
+        end
```

Comment:
Maybe we can optimize this a bit and also add an additional argument to `gamm_inc_inv_qsmall` such that the argument to the `log` in the `sqrt` expression does not have to be recomputed:
 ```suggestion
      # This deviates from the original version. The tmp variable
      # here ensures that the argument of sqrt in gamma_inc_inv_qsmall
      # is positive
      elseif !pcase && minpq < 0.02 && a < 10 && (tmp =  q*gammax(a)*sqrt(twoπ/a)) < 1 #small q
            x0 = gamma_inc_inv_qsmall(a, minpq, tmp)
      elseif abs(minpq - 0.5) < 1.0e-05
          x0 = a - 1.0/3.0 + (8.0/405.0 + 184.0/25515.0/a)/a
      elseif abs(a - 1.0) < 1.0e-4
          x0 = pcase ? -log1p(-minpq) : -log(minpq)
      elseif a < 1.0 # small value of a
          x0 = exp(logr)
      else    #large a
          haseta = true
          x0, fp = gamma_inc_inv_alarge(a, minpq, pcase)
```